File: minigpu_view/android/src/main/kotlin/dev/minigpu/view/MinigpuViewPlugin.kt
// Android stub — AHardwareBuffer/SurfaceTexture zero-copy path is not
// yet implemented.
package dev.minigpu.view

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class MinigpuViewPlugin : FlutterPlugin, MethodCallHandler {
  private lateinit var channel: MethodChannel

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(binding.binaryMessenger, "minigpu_view")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    result.error("unsupported", "Android GPU preview not yet implemented", null)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}

File: minigpu_view/ios/Classes/MinigpuViewPlugin.swift
// iOS stub — IOSurface/CVPixelBuffer zero-copy path is not yet implemented.
import Flutter
import UIKit

public class MinigpuViewPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "minigpu_view",
      binaryMessenger: registrar.messenger())
    let instance = MinigpuViewPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    result(FlutterError(
      code: "unsupported",
      message: "iOS GPU preview not yet implemented",
      details: nil))
  }
}
